consensus/participant.go

```diff
@@ -60,39 +49,19 @@ func NewParticipant(
 		option(&cfg)
 	}
 
-	// initialize forks with only finalized block.
-	// pending blocks was not recovered yet
-	forks, err := initForks(finalized, headers, updater, notifier, rootHeader, rootQC)
-	if err != nil {
-		return nil, fmt.Errorf("could not recover forks: %w", err)
-	}
-
-	// initialize the validator
-	var validator hotstuff.Validator
-	validator = validatorImpl.New(committee, forks, signer)
-	validator = validatorImpl.NewMetricsWrapper(validator, metrics) // wrapper for measuring time spent in Validator component
-
 	// get the last view we started
-	started, err := persist.GetStarted()
+	started, err := modules.Persist.GetStarted()
 	if err != nil {
 		return nil, fmt.Errorf("could not recover last started: %w", err)
 	}
 
 	// get the last view we voted
-	voted, err := persist.GetVoted()
+	voted, err := modules.Persist.GetVoted()
 	if err != nil {
 		return nil, fmt.Errorf("could not recover last voted: %w", err)
 	}
 
-	// initialize the vote aggregator
-	aggregator := voteaggregator.New(notifier, 0, committee, validator, signer)
-
-	// recover the hotstuff state, mainly to recover all pending blocks
-	// in forks
-	err = recovery.Participant(log, forks, aggregator, validator, finalized, pending)
-	if err != nil {
-		return nil, fmt.Errorf("could not recover hotstuff state: %w", err)
-	}
```

Comment:
as mentioned above, I think we should:
1. first prune the `VoteAggregator`
2. then recover the internal state of `Forks` and `VoteAggregator` here by feeding them the unfinalized blocks.